src/transformers/trainer.py

```diff
@@ -3455,7 +3453,10 @@ def _save(self, output_dir: Optional[str] = None, state_dict=None):
 
             if isinstance(self.accelerator.unwrap_model(self.model), supported_classes):
                 self.accelerator.unwrap_model(self.model).save_pretrained(
-                    output_dir, state_dict=state_dict, safe_serialization=self.args.save_safetensors
+                    output_dir,
+                    state_dict=state_dict,
+                    safe_serialization=self.args.save_safetensors,
+                    is_main_process=self.args.should_save,
                 )
             else:
```

Comment:
As there's no logic in conditional on `should_save` within this `else` branch, won't this lead to more safetensor savings unexpectedly if the model isn't a pretrained model?

Reply:
good finding, fix it.